### PR TITLE
feat(oracle): add query & cli for slashing window progress

### PR DESCRIPTION
**Pull request by adamewozniak**
Adds an oracle query for slashing window progress

closes: #1146

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/oracle/keeper/grpc_query.go [hunk 1]
```diff
@@ -136,6 +136,31 @@ func (q querier) MissCounter(
...
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	params := q.GetParams(ctx)
+
+	slashWindow := params.SlashWindow
+	votePeriod := params.VotePeriod
+	currentBlock := ctx.BlockHeight()
+	votePeriodsPerSlashWindow := slashWindow / votePeriod
+
+	currentSlashWindow := uint64(currentBlock) / votePeriodsPerSlashWindow
+	blocksIntoSlashWindow := uint64(currentBlock) - (currentSlashWindow * slashWindow)
```
Comment (adamewozniak) [inline x/oracle/keeper/grpc_query.go:157]: Taking suggestions on variable names here

Comment (RafilxTenfen) [inline x/oracle/keeper/grpc_query.go:157]:
```suggestion
	currentBlock := uint64(ctx.BlockHeight())
	votePeriodsPerSlashWindow := slashWindow / votePeriod

	currentSlashWindow := currentBlock / votePeriodsPerSlashWindow
	blocksIntoSlashWindow := currentBlock - (currentSlashWindow * slashWindow)
```

Comment (RafilxTenfen) [reply x/oracle/keeper/grpc_query.go:157]: out of ideas :/

**Review by RafilxTenfen — APPROVED** [review]

File: x/oracle/keeper/grpc_query.go (hunk 1 above)
Comment (adamewozniak) [reply x/oracle/keeper/grpc_query.go]: fec7d5b25e591cbf8c4a07f7a5a9e0cd11599f77

**Pull request merged**